### PR TITLE
Documentation updates.

### PR DESCRIPTION
**Pull request by fiveisprime**
Updated the authentication documentation to better illustrate the various ways to
authenticate in the SDK. Update the samples to use a more idiomatic JavaScript
style.

Added some more detail on how the module is organized and cleaned up the copy so
that people may more easily find the docs and get started. Fixed the formatting
of the table so that the npm install commands show up.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@fiveisprime__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (maherna). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: Documentation/Authentication.md [hunk 1]
```diff
@@ -1,140 +1,119 @@
...
+data:    Display Name:            sp-name
 data:    Service Principal Names:
-data:                             56894bd4-0fde-41d8-a0d7-5bsslccety2   <<<<<<<<<<<<<<<<<<<<< also known as clientId
-data:                             https://testap908
+data:                             **56894bd4-0fde-41d8-a0d7-5bsslccety2**
+data:                             https://sp-name
 info:    ad sp create command OK
 ```
 
-#### Step 2. Assigning a role to a ServicePrincipal by using the spn
```
Comment (balajikris) [inline Documentation/Authentication.md]: Don't you need role assignment? If you still need to do this, I'd prefer to have it in the walkthrough.

Comment (amarzavery) [inline Documentation/Authentication.md]: Without performing the role assignment the created sp cannot do anything.

File: Documentation/Authentication.md [hunk 2]
```diff
@@ -1,140 +1,119 @@
+# Authentication
```
Comment (amarzavery) [inline Documentation/Authentication.md]: The Doc had link to a script that helps you create an sp in node.js. If someone did not want to use the portal or the CLI, they can use this script to create an sp, assign a role to it and then login as that sp.
The script can be found [here](https://github.com/Azure/azure-sdk-for-node/tree/master/Documentation/ServicePrincipal). Please provide reference to this script in the walk through

File: Documentation/Authentication.md (hunk 1 above)
Comment (fiveisprime) [inline Documentation/Authentication.md]: Weird, I don't recall having to do the role assignment step. I'll get it
added back

Comment (fiveisprime) [inline Documentation/Authentication.md]: Resolved.

File: Documentation/Authentication.md (hunk 2 above)
Comment (fiveisprime) [inline Documentation/Authentication.md]: Resolved.

File: Documentation/Authentication.md [hunk 3]
```diff
@@ -1,68 +1,127 @@
...
 
-After successful login, please follow the steps mentioned below to create a serviceprincipal:
-#### Step 0. Please make sure you have selected the correct subscription. The service principal will have access to your current subscription.
-- To list all your subscriptions: `azure account list`
-- To set the subscription of your choice: `azure account set "your subscription name"`
+## Service Principal Authentication
+
+Interactive login, similar to how the CLI authenticates, is the easiest way to
+authenticate; however, when using the Node.js SDK programmatically, you will need
+to use service principal authentication. This essentially creates keys for your
```
Comment (amarzavery) [inline Documentation/Authentication.md]: > however, when using the Node.js SDK programmatically, you **will need to use service principal authentication**

It is not a requirement to use service principal authentication. The runtime supports all the 3 forms of authentication. The user may choose to use whatever seems fit for his scenario.

Comment (fiveisprime) [inline Documentation/Authentication.md]: I was trying to capture the intent of this sentence:

>One does not want to login interactively all the time. Azure provides service principal authentication as a secure way for silent login.

I'll see if I can make that clearer.

Comment (amarzavery) [inline Documentation/Authentication.md]: I see. It wasn't clear in the first go. The updated commit LGTM.

**Pull request merged**